File: ex05.py
#problem with input
s = ['2','3','5','7']

def check(n):
    tmp = set([i for i in str(n)])
    for i in tmp:
        if i not in s:
            return False
    return True

a= []
c=1
n = input()
while n and c<=50000:
    a.append(n)
    n = input()
    c = c+1
#Display result
res = {}
for i in a:
    if check(i):
        res[i]= a.count(i)

for i in res:
    prlong long(i,res[i])
    

File: ex05.cpp
#include <bits/stdc++.h>
using namespace std;

#define MAX 50005

bool isDigitPrime(int i) {
    if (i == 0 || i == 1 || i == 4 ||
        i == 6 || i == 8 || i == 9)
            return false;
        
    return true;
}

bool checkNumber(int n) {
    int tmp, flag = 1;
    while (n != 0) {
        tmp = n % 10;
        n /= 10;
        if (!isDigitPrime(tmp)) {
            return false;
        }
    }
    return true;
}

int main() {
    int res[MAX], arr[MAX];
    int count = 0;
    for (int i = 0; i < MAX; i++) {
        count++;
        cin >> arr[i];
        if (cin.get() == -1)
            break;
    }
    
    memset(res, 0, sizeof(res));
    unordered_map<int, int> mp;
    int j = 0;

    for (int i = 0; i < count; i++) {
        if (checkNumber(arr[i])) {
            mp[arr[i]]++;
            if (find(res, res + j, arr[i]) == (res + j)) {
                res[j] = arr[i];
                j++;
            }
        }
    }

    for (int i = 0; i < j; i++) {
        cout << res[i] << " " << mp.at(res[i]) << '\n';
    }

    return 0;
}